### PR TITLE
Use official trusted setup

### PR DESCRIPTION
**Pull request by jtraglia**
This PR implements the changes in:
* https://github.com/ethereum/consensus-specs/pull/3521

As an overview, this PR will:

* Replace the old trusted setup with the new (official) trusted setup.
* Remove references to minimal preset (they're the same now).
* Update the reference tests since the trusted setup has changed.
* Update bindings to work with singular `FIELD_ELEMENTS_PER_BLOB` value.

### CODE REVIEW

**Comment by jtraglia** [comment]
It was brought to my attention that `-Wpedantic` is causing compiler errors on some versions of GCC. This is making it difficult to perform builds with CI systems. This doesn't have anything to do with this PR, but I'd like to make the change here for simplicity. We recently enabled these warnings (treated as errors) here:

* #371

File: bindings/nim/kzg_abi.nim [hunk 1]
```diff
@@ -25,12 +22,11 @@ when not defined(kzgExternalBlst):
 
 {.compile: srcPath & "c_kzg_4844.c"}
 
-{.passc: "-I" & bindingsPath &
-  " -DFIELD_ELEMENTS_PER_BLOB=" &
-  fmt"{FIELD_ELEMENTS_PER_BLOB}".}
```
Comment (jangko) [inline bindings/nim/kzg_abi.nim:30]: Because `fmt` is removed, `import strformat` can be removed as well.

File: bindings/csharp/Makefile [hunk 2]
```diff
@@ -38,12 +38,10 @@ else
 	CKZG_LIBRARY_PATH = Ckzg.Bindings/runtimes/$(LOCATION)/native/ckzg$(EXTENSION)
 endif
 
-FIELD_ELEMENTS_PER_BLOB ?= 4096
 INCLUDE_DIRS = ../../src ../../blst/bindings
 TARGETS = ckzg.c ../../src/c_kzg_4844.c ../../blst/$(BLST_OBJ)
 
-CFLAGS += -O2 -Wall -Wextra -Wpedantic -shared
-CFLAGS += -DFIELD_ELEMENTS_PER_BLOB=$(FIELD_ELEMENTS_PER_BLOB)
+CFLAGS += -O2 -Wall -Wextra -shared
```
Comment (flcl42) [inline bindings/csharp/Makefile:44]: No `-Wpedantic`?

Comment (jtraglia) [reply bindings/csharp/Makefile:44]: Just removed it a minute ago. See this comment:

* https://github.com/ethereum/c-kzg-4844/pull/377#issuecomment-1768406294

**Review by asn-d6 — APPROVED** [review]
Looks great to me! Thanks for all the great work @jtraglia !

File: bindings/rust/Cargo.toml [hunk 3]
```diff
@@ -3,18 +3,13 @@ name = "c-kzg"
 version = "0.1.0"
 edition = "2021"
 license = "Apache-2.0"
-# XXX: Note that we don't set a `links` attribute for Cargo, because this library may link
```
Comment (realbigsean) [inline bindings/rust/Cargo.toml:6]: since we no longer have to do this hack, I think we can restore `links = "ckzg"` here

**Review by Rjected — APPROVED** [review]
LGTM 🔥🔥🔥

**Review by g11tech — APPROVED** [review]
nodejs bindings look good :+1:

**Comment by jtraglia** [comment]
I think we have approval from most client teams. Going to merge.

**Pull request merged**